command/v6/enable_service_access_command_test.go

```diff
@@ -0,0 +1,246 @@
+	"code.cloudfoundry.org/cli/command/flag"
+	"code.cloudfoundry.org/cli/command/translatableerror"
+	. "code.cloudfoundry.org/cli/command/v6"
+	"code.cloudfoundry.org/cli/command/v6/v6fakes"
+	"code.cloudfoundry.org/cli/util/configv3"
+	"code.cloudfoundry.org/cli/util/ui"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+)
+
+var _ = Describe("enable-service-access command", func() {
+	var (
+		cmd             EnableServiceAccessCommand
+		testUI          *ui.UI
+		fakeConfig      *commandfakes.FakeConfig
+		fakeSharedActor *commandfakes.FakeSharedActor
+		fakeActor       *v6fakes.FakeEnableServiceAccessActor
+		binaryName      string
+		executeErr      error
+		extraArgs       []string
+	)
+
+	BeforeEach(func() {
+		testUI = ui.NewTestUI(nil, NewBuffer(), NewBuffer())
+		fakeConfig = new(commandfakes.FakeConfig)
+		fakeSharedActor = new(commandfakes.FakeSharedActor)
+		fakeActor = new(v6fakes.FakeEnableServiceAccessActor)
+		extraArgs = nil
+
+		cmd = EnableServiceAccessCommand{
+			UI:          testUI,
+			Config:      fakeConfig,
+			SharedActor: fakeSharedActor,
+			Actor:       fakeActor,
+			RequiredArgs: flag.Service{
+				Service: "some-service",
+			},
+		}
+
+		binaryName = "faceman"
+		fakeConfig.BinaryNameReturns(binaryName)
+	})
+
+	JustBeforeEach(func() {
+		executeErr = cmd.Execute(extraArgs)
+	})
+
+	When("the user provides arguments", func() {
+		BeforeEach(func() {
+			extraArgs = []string{"some-extra-arg"}
+		})
+
+		It("fails with a TooManyArgumentsError", func() {
+			Expect(executeErr).To(MatchError(translatableerror.TooManyArgumentsError{
+				ExtraArgument: "some-extra-arg",
+			}))
+		})
+	})
+	When("a cloud controller API endpoint is set", func() {
+		BeforeEach(func() {
+			fakeConfig.TargetReturns("some-url")
+		})
+
+		When("checking target fails", func() {
+			BeforeEach(func() {
+				fakeSharedActor.CheckTargetReturns(actionerror.NotLoggedInError{BinaryName: binaryName})
+			})
+
+			It("returns a not logged in error", func() {
+				Expect(fakeSharedActor.CheckTargetCallCount()).To(Equal(1))
+				Expect(executeErr).To(MatchError(actionerror.NotLoggedInError{BinaryName: binaryName}))
+			})
+		})
+
+		When("the user is logged in", func() {
+			BeforeEach(func() {
+				fakeConfig.CurrentUserReturns(
+					configv3.User{Name: "admin"},
+					nil)
+				fakeSharedActor.IsLoggedInReturns(true)
+			})
+
+			When("no flags are passed", func() {
+				BeforeEach(func() {
+					fakeActor.EnableServiceForAllOrgsReturns(nil, nil)
+				})
+
+				It("passes on the service", func() {
+					Expect(fakeActor.EnableServiceForAllOrgsCallCount()).To(Equal(1))
+					service := fakeActor.EnableServiceForAllOrgsArgsForCall(0)
+					Expect(service).To(Equal("some-service"))
+				})
+
+				It("displays informative success message", func() {
+					Expect(testUI.Out).To(Say("Enabling access to all plans of service some-service for all orgs as admin\\.\\.\\."))
```

Comment:
FYI, you can use [regexp.QuoteMeta](https://golang.org/pkg/regexp/#QuoteMeta) to not need to escape those `.` characters. But this is such minor escaping (and we escape dots like this in plenty of places), so I don't have a problem with this staying as is.

Reply:
(And in this case since the characters needed to write `regexp.QuoteMeta` is more than the extra characters needed for the escaping, it feels especially unnecessary.)